pennylane/io.py

```diff
@@ -410,113 +407,131 @@ def from_qiskit_op(qiskit_op, params=None, wires=None):
         raise RuntimeError(_MISSING_QISKIT_PLUGIN_MESSAGE) from e
 
 
-def from_qasm(quantum_circuit: str, measurements=False):
-    """Loads quantum circuits from a QASM string using the converter in the
+def from_qasm(quantum_circuit: str, measurements=None):
+    """
+    Loads quantum circuits from a QASM string using the converter in the
     PennyLane-Qiskit plugin.
 
     Args:
         quantum_circuit (str): a QASM string containing a valid quantum circuit
         measurements (None | MeasurementProcess | list[MeasurementProcess]): an optional PennyLane
-            measurement or list of PennyLane measurements that overrides any terminal measurements
-            that may be present in the input circuit. If set to ``None``, existing measurements
-            in the input circuit will be used.
+            measurement or list of PennyLane measurements that overrides the terminal measurements
+            that may be present in the input circuit. Defaults to ``None``, such that existing measurements
+            in the input circuit are kept. Mid-circuit measurements are kept either way.
 
     Returns:
-        function: the PennyLane template created based on the QASM string
+        function: the PennyLane quantum function created based on the QASM string. This function itself returns the mid-circuit measurements plus the terminal measurements by default (``measurements=None``), and returns **only** the measurements from the ``measurements`` argument otherwise.
 
-    By default, ``from_qasm`` will remove any measurements that are present in the QASM code.
-    If the QASM code contains measurements, set ``measurements=None`` to keep them in the
-    output of ``from_qasm``.
+    **Example:**
 
-    .. warning::
+    .. code-block:: python
 
-        The current default behaviour of removing measurements in the QASM code is deprecated
-        and will be changed in a future release. Starting in version ``0.38``, ``from_qasm``
-        will keep the measurements from the QASM code by default. To remove all measurements,
-        set ``measurements=[]`` which overrides the existing measurements with an empty list.
+        qasm_code = 'OPENQASM 2.0;'\\
+                    'include "qelib1.inc";'\\
+                    'qreg q[2];'\\
+                    'creg c[2];'\\
+                    'h q[0];'\\
+                    'measure q[0] -> c[0];'\\
+                    'rz(0.24) q[0];'\\
+                    'cx q[0], q[1];'\\
+                    'measure q -> c;'
 
-    **Example:**
+        loaded_circuit = qml.from_qasm(qasm_code)
+
+    >>> print(qml.draw(loaded_circuit)())
+    0: ──H──┤↗├──RZ(0.24)─╭●──┤↗├─┤
+    1: ───────────────────╰X──┤↗├─┤
+
+    Calling the quantum function returns a tuple containing the mid-circuit measurements and the terminal measurements.
+
+    >>> my_circuit()
```

Comment:
`my_circuit` needs to be defined 🤔

Reply:
My bad 😳. This is supposed to be `loaded_circuit`, thank you for spotting this!!